src/librustc_codegen_llvm/back/linker.rs

```diff
@@ -895,14 +885,12 @@ impl<'a> Linker for EmLinker<'a> {
 }
 
 fn exported_symbols(tcx: TyCtxt, crate_type: CrateType) -> Vec<String> {
-    let mut symbols = Vec::new();
-
     let export_threshold = symbol_export::crates_export_threshold(&[crate_type]);
-    for &(symbol, level) in tcx.exported_symbols(LOCAL_CRATE).iter() {
-        if level.is_below_threshold(export_threshold) {
-            symbols.push(symbol.symbol_name(tcx).to_string());
-        }
-    }
+
+    let mut symbols = tcx.exported_symbols(LOCAL_CRATE).iter()
+        .filter(|(_, level)| level.is_below_threshold(export_threshold))
+        .map(|(symbol, _)| symbol.symbol_name(tcx).to_string())
+        .collect::<Vec<_>>();
```

Comment:
There's no actual advantage to using iterators here I expect because filter reduces the lower size hint to 0.

Since I think the for loop is slightly more readable, can you revert to that?

Reply:
Oof; you're right, I forgot about that.